sceptre/plan/actions.py

```diff
@@ -1000,3 +1000,115 @@ def diff(self, stack_differ):
         :rtype: sceptre.diffing.stack_differ.StackDiff
         """
         return stack_differ.diff(self)
+
+    @add_stack_hooks
+    def detect_stack_drift(self):
+        """
+        Detects stack drift for a running stack.
+
+        :returns: The stack drift.
+        :rtype: Tuple[str, Union[str, dict]]
+        """
+        response = self._detect_stack_drift()
+
+        detection_id = response["StackDriftDetectionId"]
+        status = self._wait_for_drift(detection_id)
+
+        if status == "DETECTION_COMPLETE":
+            response = self._describe_stack_resource_drifts()
+            return_value = (self.stack.external_name, response)
+        else:
+            return_value = (self.stack.external_name, status)
+
+        return return_value
+
+    def _wait_for_drift(self, detection_id):
+        """
+        Waits for drift detection to complete.
+
+        :param detection_id: The drift detection ID.
+        :type detection_id: str
+
+        :returns: The drift status.
+        :rtype: str
+        """
+        timeout = 300
+        elapsed = 0
+
+        while True:
+            if elapsed >= timeout:
+                return "TIMED_OUT"
+
+            self.logger.debug("%s - Waiting for drift detection", self.stack.name)
+            response = self._describe_stack_drift_detection_status(detection_id)
+
+            status = response["DetectionStatus"]
+            self._print_drift_status(response)
+
+            if status == "DETECTION_IN_PROGRESS":
+                time.sleep(10)
+                elapsed += 10
+            else:
+                return status
```

Comment:
You haven't accounted for `"DETECTION_FAILED"` which is a possible status that could be returned. Based upon this code, it appears as if, in the case where detection fails, you would just pick up the results of the last successful drift detection and then return that as if nothing wrong had happened.

Reply:
@jfalkenstein Good pickup. What it was doing was returning `DETECTION_FAILED` and no drift info. What it should have done is returned `DETECTION_FAILED` and the response from AWS, since `DETECTION_FAILED` can mean some drift was detected but detection then failed, preventing all of it being detected.

Reply:
It's still not clear to me why you wouldn't want to raise an error if detection fails, although I don't really understand the cases (other than AWS failure) that would lead to DETECTION_FAILED.